### ISSUE TITLE
Add support for unsigned types.

### ISSUE THREAD

**Issue by Nicolas-GUILLAUME** [awslabs/smithy-rs#1836]
We have a use case where our system mostly uses unsigned types. 

This forces us to covert all our unsigned types into signed types. So far we've made it work but this will become a problem for us at some point.

If "native" support is not an option it would be nice to have a way to do it via a custom code generator.

**Comment by jdisanti** [comment]
It seems like it should be possible to implement this as a plugin to smithy-rs, but the `RustCodegenDecorator` doesn't appear to have a way to augment the `SymbolProvider` implementation.

It doesn't look like adding this support should be too difficult. The symbol provider is currently resolved [from the root plugin](https://github.com/awslabs/smithy-rs/blob/917d0a62b50bd03734747648b87ecb665e213c6a/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/CodegenVisitor.kt#L82). We should be able to add a method to `RustCodegenDecorator` that takes a base symbol provider as an argument, and returns an updated/revised symbol provider, and then call that in `CodegenVisitor`. I think that should unblock this use case, although there may be other snags we aren't aware of yet.

**Comment by david-perez** [comment]
There was some discussion on whether `@range(min: 0)` should generate unsigned integers [in this thread](https://github.com/awslabs/smithy-rs/pull/1199#discussion_r809317248). There's an unresolved question of what to do with e.g. `u32` values that do not fit within what the Smithy spec says the range of the `integer` shape is.
